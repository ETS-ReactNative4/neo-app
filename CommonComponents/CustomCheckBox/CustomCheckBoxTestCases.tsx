import React, { useState, useEffect } from "react";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import CustomCheckBox, {
  ICheckBoxData,
  ISuggestedDetails
} from "./CustomCheckBox";
import { StyleSheet, View } from "react-native";

const data: ICheckBoxData[] = [
  {
    text: "Lorem Ipsum"
  },
  {
    text: "Lorem Ipsum"
  },
  {
    text: "Lorem Ipsum"
  },
  {
    text: "Lorem Ipsum"
  },
  {
    text: "Lorem Ipsum"
  }
];

interface CheckBoxWrapperProps {
  checkboxData: ICheckBoxData[];
}

const styles = StyleSheet.create({
  checkboxWrapper: {
    padding: 40
  }
});

const CheckBoxWrapper = ({ checkboxData }: CheckBoxWrapperProps) => {
  const [suggestedDetails, setSuggestedDetails] = useState<ISuggestedDetails[]>(
    []
  );

  useEffect(() => {
    setSuggestedDetails(
      checkboxData.map((checkboxDataObj, dataIndex) => {
        return {
          index: dataIndex,
          text: checkboxDataObj.text,
          isChecked: false
        };
      })
    );
  }, [checkboxData]);

  const selectSuggestedDetails = (dataIndex: number) => {
    const suggestedList = [...suggestedDetails];
    suggestedList[dataIndex].isChecked = !suggestedList[dataIndex].isChecked;
    setSuggestedDetails(suggestedList);
  };

  return (
    <View style={styles.checkboxWrapper}>
      {suggestedDetails.map((dataObj, index) => {
        const onSelect = () => {
          selectSuggestedDetails(dataObj.index);
        };

        return (
          <CustomCheckBox
            key={index}
            isChecked={dataObj.isChecked}
            action={onSelect}
            text={dataObj.text}
          />
        );
      })}
    </View>
  );
};

const CustomCheckBoxTestCases: ITestCase[] = [
  {
    title: "Check box",
    Component: <CheckBoxWrapper checkboxData={data} />
  }
];

export default CustomCheckBoxTestCases;
