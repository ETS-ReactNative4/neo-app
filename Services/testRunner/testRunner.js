import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
