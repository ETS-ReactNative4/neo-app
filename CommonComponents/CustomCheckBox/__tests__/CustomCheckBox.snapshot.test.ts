import CustomCheckBoxTestCases from "../CustomCheckBoxTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

CustomCheckBoxTestCases.map(testRunner);
