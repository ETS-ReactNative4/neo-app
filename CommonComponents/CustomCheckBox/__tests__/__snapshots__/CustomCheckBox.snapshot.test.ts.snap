// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check box 1`] = `
<View
  style={
    Object {
      "padding": 40,
    }
  }
/>
`;
